### PR TITLE
Refactor Artifacts service

### PR DESCRIPTION
**Pull request by seaneganx**
* Move model storage code into DAO
* Use names package for parsing and representing resource names
* Add test suite

### CODE REVIEW

**Comment by seaneganx** [comment]
Fixes #92 #90 #61

File: server/actions_artifacts_test.go [hunk 1]
```diff
@@ -0,0 +1,894 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
```
Comment (timburks) [inline server/actions_artifacts_test.go:1]: nit: 2021

**Review by timburks — APPROVED** [review]
This looks great! Thanks for staging it in multiple commits, that made it much easier to digest.

File: server/actions_artifacts_test.go (hunk 1 above)
Comment (seaneganx) [reply server/actions_artifacts_test.go]: Fixed.

**Pull request merged**